### ISSUE TITLE
[Bug]: Profile freezes Firefly v1.3.3

### ISSUE THREAD

**Issue by kraftjs** [iotaledger/firefly#2345]
### Task description

_Creating this issue to provide a description of the problem and consolidate related bugs._
Please discuss findings related to the bug in comments below.

#### Description:

This bug appears isolated to the ~~Windows platform~~ (now reported on Windows and macOS) and Firefly version 1.3.3. Users experiencing the bug receive an error message: `Error Type: Callback Error latestAddress`, or their Firefly application freezes after selecting a profile and entering their PIN.

I reproduced the bug on Windows 10 with Firefly v1.3.3 by sending a transaction to an address generated by a new Ledger profile. There was no error message, but my Firefly application froze (impossible to run Diagnostics or the Error Log). I could exit the application without the task manager. Creating a profile and restoring the wallet made the new balance visible after Firefly sync'd, but then Firefly immediately froze again. I am able to use non-Ledger profiles on Firefly, but then Firefly freezes when trying to switch to a Ledger profile after inputting the PIN.

#### Actions that cause Firefly to freeze or emit an error:
- Sending a transaction to the address generated by the Ledger profile.
- Staking funds already on the Ledger profile.
- Changing to different wallets within the Ledger profile (see Issue 2318 below)
    - starting at wallet _0_ doesn't cause the bug
    - switching to wallet _1_ causes the bug
    - switching to wallet _2_ doesn't cause the bug

#### Related bug reports:
- Issues mentioning `Error Type: Callback Error latestAddress`
    - Issue #2198
    - Issue #2319
    - Issue #2326 
- Issue #2341 
- Issue #2343
- Issue #2370

#### Potentially Related Bugs:
- Issue #2318 

### Requirements

- [x] See if bug is replicable in `develop` branch
- [ ] See if bug is replicable in older version of Firefly
- [x] See if bug is replicable on Mac

### Acceptance criteria

- [ ] Implement hotfix to solve bug.

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

**Comment by kraftjs** [comment]
Bug is still present in `develop` branch. I was able to sync funds and generate a receiving address on the Ledger profile without freezing, but switching to a non-ledger profile and sending the funds to that generated address and then switching back to the Ledger profile caused Firefly to freeze. Exiting the application and trying to re-open the Ledger profile causes Firefly to freeze at loading screen.

**Comment by kraftjs** [comment]
Bug seems variable. As I changed versions and tried to replicate it, it would appear on a version and then later after switching to other versions and back not re-appear. Other times, Firefly wouldn't freeze but the balance displayed was wrong, even after forcing a sync several times.

**Firefly v1.3.0 (Windows)** - No issues. Can send and receive transactions in both Ledger profile and non-Ledger profile. Couldn't access profile at first since it was still getting stuck at loading screen after testing version 1.3.3. Deleting the profile from C:\Users\<username>\AppData\Roaming\Firefly\__storage__ and recovering fixed the problem.
<br />

**Firefly v1.3.1 (Windows)** - No issues. Can send and receive transactions in both Ledger profile and non-Ledger profile. Didn't need to delete faulty profiles from C:\Users\<username>\AppData\Roaming\Firefly\__storage__ since I tested v1.3.0 before this and it didn't have the bug.
<br />

**Firefly v1.3.2 (Windows)** - Intermittent Issues. Sent a transaction from Ledger profile to non-Ledger profile, which resulted in a freeze at the loading screen for _non-Ledger_ profile after inputting PIN. Deleting profile from __storage__ and recovering still resulted in a freeze as soon as I return to wallet view. But then after trying Firefly v1.3.0 and v1.3.1, when I returned to v1.3.2 I have been unable to reproduce the bug.
<br />

**Firefly v1.3.3 (Windows)** - Unable to replicate Firefly freezing after doing the testing above, but wallet balances are wrong (Firefly still warns if I try to send the incorrect balance). Sometimes takes 10+ syncs before balance is corrected. Then when I log back into the wallet again the balance is back to being incorrect.
